server/server/src/billing/stripe.rs

```diff
@@ -0,0 +1,171 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum StripeResult<U> {
+    Ok(U),
+    Err(StripeErrorContainer),
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct StripeErrorContainer {
+    pub error: StripeError,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct StripeError {
+    #[serde(rename = "type")]
+    pub error_type: StripeErrorType,
+    pub code: StripeErrorCode<StripeKnownErrorCode>,
+    pub decline_code: Option<StripeErrorCode<StripeKnownErrorDeclineCode>>,
+    pub doc_url: String,
+    pub message: String,
+    pub param: String,
+    pub payment_method_type: Option<String>,
+    pub charge: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum StripeErrorType {
+    #[serde(rename = "api_error")]
+    ApiError,
+    #[serde(rename = "card_error")]
+    CardError,
+    #[serde(rename = "idempotency_error")]
+    IdempotencyError,
+    #[serde(rename = "invalid_request_error")]
+    InvalidRequestError,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(untagged)]
+pub enum StripeErrorCode<E> {
+    Known(E),
+    Unknown(String),
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum StripeKnownErrorCode {
+    CardDeclineRateLimitExceeded,
+    CardDeclined,
+    ExpiredCard,
+    IncorrectCvc,
+    IncorrectNumber,
+    InsufficientFunds,
+    InvalidCvc,
+    InvalidExpiryMonth,
+    InvalidExpiryYear,
+    InvalidNumber,
+    PaymentIntentAuthenticationFailure,
+    ProcessingError,
+    SetupIntentAuthenticationFailure,
+}
+// customer_max_payment_methods
+
+#[derive(Debug, Serialize, Deserialize)]
```

Comment:
you should add a comment linking to stripe docs for where all these came from, also why are some commented out?